crates/task-impls/src/quorum_vote.rs

```diff
@@ -0,0 +1,359 @@
+                    );
+                    self.vote_dependencies.insert(
+                        view,
+                        async_spawn(async move {
+                            dependency.completed().await;
+                            broadcast_event(HotShotEvent::ViewChange(view + 1), &event_sender)
+                                .await;
+                            broadcast_event(HotShotEvent::DummyQuorumVoteSend(view), &event_sender)
+                                .await;
+                        }),
+                    );
+                }
+            }
+            HotShotEvent::VidDisperseRecv(disperse, _sender) => {
+                let view = disperse.data.get_view_number();
+
+                // stop polling for the received disperse after verifying it's valid
+                self.quorum_network
+                    .inject_consensus_info(ConsensusIntentEvent::CancelPollForVIDDisperse(
+                        *disperse.data.view_number,
+                    ))
+                    .await;
+
+                if self.vote_dependencies.get(&view).is_none() {
+                    if !validate_vid(disperse, event_sender.clone()) {
+                        return;
+                    }
+                    let dependency = self.create_vote_dependency(
+                        &VoteDependency::Vid,
+                        view,
+                        event_sender.clone(),
+                    );
+                    self.vote_dependencies.insert(
+                        view,
+                        async_spawn(async move {
+                            dependency.completed().await;
+                            broadcast_event(HotShotEvent::ViewChange(view + 1), &event_sender)
+                                .await;
+                            broadcast_event(HotShotEvent::DummyQuorumVoteSend(view), &event_sender)
+                                .await;
+                        }),
+                    );
+                }
+            }
+            HotShotEvent::ViewChange(new_view) => {
+                debug!("View Change event for view {} in consensus task", *new_view);
+
+                let old_view_number = self.next_vote_view;
+
+                // Start polling for VID disperse for the new view
+                self.quorum_network
+                    .inject_consensus_info(ConsensusIntentEvent::PollForVIDDisperse(
+                        *old_view_number + 1,
+                    ))
+                    .await;
+
+                // update the view in state to the one in the message
+                // Publish a view change event to the application
+                if !self.update_next_vote_view(new_view).await {
+                    debug!("view not updated");
+                    return;
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TaskState for QuorumVoteTaskState<TYPES, I> {
+    type Event = HotShotEvent<TYPES>;
```

Comment:
Checking my understanding -- Is the `Arc` needed for memory efficiency or thread safety? Since you mentioned "have to" I assume it's the latter, but I'm not sure why it's a problem to not use `Arc`.

Reply:
Oh just realized events were all wrapped inside `Arc` in other tasks. I'll update this.